src/redux/books/booksSlice.js

```diff
@@ -1,39 +1,82 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 
-const initialState = [
-  {
-    id: 'item1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
-  },
-  {
-    id: 'item2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    id: 'item3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  },
-];
+const baseURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/NWL6J2bcQLwcLHk7uDm7/books';
+
+const initialState = {
+  books: [],
+  isLoading: false,
+};
+
+export const fetchBooks = createAsyncThunk('books/fetchBooks', async () => {
+  try {
+    const response = await axios.get(baseURL);
+    return response.data;
+  } catch (error) {
+    throw new Error('Failed to fetch books');
+  }
+});
+
+export const addBook = createAsyncThunk('books/addBook', async (newBook) => {
+  try {
+    await axios.post(baseURL, newBook);
+  } catch (error) {
+    console.error('Error posting book:', error.response.data);
+    throw error;
+  }
+});
+
+export const removeBook = createAsyncThunk('books/removeBook', async (item_id) => {
+  try {
+    await axios.delete(`${baseURL}/${item_id}`);
+  } catch (error) {
+    console.error('Error removing book:', error);
+    throw error;
+  }
+});
 
```

Comment:
- [ ] Great work so far on making the changes as requested 👍 but it would be great if you make your app work as expected in the project requirements 🤔 because when we add or remove a book then we need to refresh the page to render the books from the API 😨 Kindly make your app behave like a Single Page Application because Single Page Application makes all kinds of API calls and renders data without refreshing your HTML page 😊